storage/acl.go

```diff
@@ -91,8 +90,8 @@ func (a *ACLHandle) Delete(ctx context.Context, entity ACLEntity) (err error) {
 
 // Set sets the role for the given entity.
 func (a *ACLHandle) Set(ctx context.Context, entity ACLEntity, role ACLRole) (err error) {
-	ctx = trace.StartSpan(ctx, "cloud.google.com/go/storage.ACL.Set")
-	defer func() { trace.EndSpan(ctx, err) }()
+	ctx, _ = startSpan(ctx, "storage.ACL.Set")
```

Comment:
Yes, we will return the span and add more context pieces wherever applicable. The ability to return a span is one of the main changes for the new trace package.

Error recording is also handled in `endSpan` rather than at each individual callsite. I think this will help reduce the boilerplate as well.